### PR TITLE
Added endpoint sign in

### PR DESCRIPTION
**Pull request by edilsonLon**
## Summary

- ### I have installed the jest-extended module
- ### I have created sign in endpoint
- ### I did the tests for endpoint sign in

## Known Issues

- ### None
## Trello Card

- ### https://trello.com/c/QLu6L6Xp/2-sign-in

### CODE REVIEW

File: app/errors.js [hunk 1]
```diff
@@ -10,5 +10,7 @@ exports.DEFAULT_ERROR = 'default_error';
 exports.defaultError = message => internalError(message, exports.DEFAULT_ERROR);
 
 exports.BAD_REQUEST_ERROR = 'bad_request';
-
 exports.badRequest = message => internalError(message, exports.BAD_REQUEST_ERROR);
+
+exports.UNAUTHORIZED_ERROR = 'unauthorized_request';
```
Comment (martinremis) [inline app/errors.js]: This should be an authentication error, not an authorization error (by the way you use it in the controller)

File: app/controllers/users.controller.js [hunk 2]
```diff
@@ -21,3 +21,22 @@ exports.signUp = async (req, res, next) => {
...
+
+exports.signIn = async (req, res, next) => {
+  const { body } = req;
+  const { email, password } = userSignInMapper(body);
+  try {
+    const user = await userFindByEmail(email);
+    if (!user) throw errors.unauthorized(EMAIL_OR_PASSWORD_DO_NOT_MATCH);
+    const comparePassword = compareHash(user.password, password);
+    if (!comparePassword) throw errors.unauthorized(EMAIL_OR_PASSWORD_DO_NOT_MATCH);
+    const token = jwtEncode({
```
Comment (martinremis) [inline app/controllers/users.controller.js]: I think you could move this to the helper with a method like encodeJwtUser()

File: app/middlewares/errors.js [hunk 3]
```diff
@@ -6,7 +6,8 @@ const DEFAULT_STATUS_CODE = 500;
 const statusCodes = {
   [errors.DATABASE_ERROR]: 503,
   [errors.DEFAULT_ERROR]: 500,
-  [errors.BAD_REQUEST_ERROR]: 400
+  [errors.BAD_REQUEST_ERROR]: 400,
+  [errors.UNAUTHORIZED_ERROR]: 401
```
Comment (martinremis) [inline app/middlewares/errors.js]: Unauthorized use to be 403 (aka forbidden). 401 use to be authentication error.

**Review by martinremis — CHANGES_REQUESTED** [review]
Well done 🎉 😄 
Just a few comments, specially about authentication vs authorization

File: app/errors.js (hunk 1 above)
Comment (edilsonLon) [reply app/errors.js]: Perfect, I have changed it by request forbidden.

File: app/middlewares/errors.js (hunk 3 above)
Comment (edilsonLon) [reply app/middlewares/errors.js]: I have created a forbidden error with the code 403. 👀

File: app/controllers/users.controller.js (hunk 2 above)
Comment (edilsonLon) [reply app/controllers/users.controller.js]: jwtEncode is currently in the helper folder ([app/helpers/auth.js](https://github.com/wolox-training/el-express-js/pull/7/files#diff-3cd07146227b248bec032c8f6c0b5a5df00d22b99c7620882f97c34cfbc32030) ). I have changed the payload for the serialized user.

File: app/middlewares/errors.js [hunk 4]
```diff
@@ -6,7 +6,9 @@ const DEFAULT_STATUS_CODE = 500;
 const statusCodes = {
   [errors.DATABASE_ERROR]: 503,
   [errors.DEFAULT_ERROR]: 500,
-  [errors.BAD_REQUEST_ERROR]: 400
+  [errors.BAD_REQUEST_ERROR]: 400,
+  [errors.UNAUTHORIZED_ERROR]: 401,
+  [errors.AUTHENTICATION_ERROR]: 403
```
Comment (martinremis) [inline app/middlewares/errors.js]: AUTHENTICATION_ERROR should be 401 and UNAUTHORIZED_ERROR (forbidden is better) 403 😛

**Review by martinremis — CHANGES_REQUESTED** [review]

File: app/middlewares/errors.js (hunk 4 above)
Comment (edilsonLon) [reply app/middlewares/errors.js]: Sorry. I understand now, I have replaced it.😅😅😅😅

**Review by martinremis — APPROVED** [review]

**Pull request merged**